### ISSUE TITLE
Rails 5 compatibility

### ISSUE THREAD

**Issue by patbl** [camertron/generated-assets#4]
There's a dependency on Railties 4 or earlier. Rails 5 support would be much appreciated!

**Comment by camertron** [comment]
Hey @patbl thanks for filing an issue. I just released v2.1.0 of generated-assets which should work with Rails 5. Can you give it a try and let me know if it works for you?

Also I'm curious - what are you using generated-assets for? Are you also using i18n-js-assets or do you depend on generated-assets directly? Thanks!

**Comment by patbl** [comment]
Yes, will do!

I'm pretty sure we're using it only for i18n-js-assets. The two gems are listed separately in our Gemfile, but I think that's because of something to do with [this](https://collectiveidea.com/blog/archives/2016/10/06/bundlers-multiple-source-security-vulnerability/). I noticed i18n-js-assets after I posted this issue, and I didn't want to be annoying and post two issues about the same thing.

I'll probably be able to upgrade the gems within a week. I'll let you know how it goes. Thanks for updating them so quickly!

**Comment by patbl** [comment]
I tried upgrading i18n-js-assets to 2.1.0 (with generated-assets 2.1.0). There weren't any errors, but it didn't reload translations, even after a server restart. That is, if I modified a `config/locales` file and restarted the server, I would still get the old translation in development.

So I downgraded to i18n-js-assets 1.1.1 (with generated-assets 1.3.0). Everything worked fine, until I tried upgrading Rails from 4.2.7.1 to 4.2.8. Then I got this error when I tried to run any Rails command:

```
bundle exec rake db:create db:structure:load
rake aborted!
TypeError: no implicit conversion of nil into String
/home/ubuntu/academia-app/vendor/bundle/ruby/2.2.0/gems/generated-assets-1.1.1/lib/generated-assets/railtie.rb:7:in `join'
/home/ubuntu/academia-app/vendor/bundle/ruby/2.2.0/gems/generated-assets-1.1.1/lib/generated-assets/railtie.rb:7:in `block in <class:Railtie>'
/home/ubuntu/academia-app/vendor/bundle/ruby/2.2.0/gems/activesupport-4.2.8/lib/active_support/lazy_load_hooks.rb:36:in `call'
/home/ubuntu/academia-app/vendor/bundle/ruby/2.2.0/gems/activesupport-4.2.8/lib/active_support/lazy_load_hooks.rb:36:in `execute_hook'
/home/ubuntu/academia-app/vendor/bundle/ruby/2.2.0/gems/activesupport-4.2.8/lib/active_support/lazy_load_hooks.rb:45:in `block in run_load_hooks'
/home/ubuntu/academia-app/vendor/bundle/ruby/2.2.0/gems/activesupport-4.2.8/lib/active_support/lazy_load_hooks.rb:44:in `each'
/home/ubuntu/academia-app/vendor/bundle/ruby/2.2.0/gems/activesupport-4.2.8/lib/active_support/lazy_load_hooks.rb:44:in `run_load_hooks'
/home/ubuntu/academia-app/vendor/bundle/ruby/2.2.0/gems/railties-4.2.8/lib/rails/application.rb:93:in `inherited'
/home/ubuntu/academia-app/config/application.rb:76:in `<module:Academia>'
/home/ubuntu/academia-app/config/application.rb:59:in `<top (required)>'
/home/ubuntu/academia-app/Rakefile:12:in `require'
/home/ubuntu/academia-app/Rakefile:12:in `<top (required)>'
/home/ubuntu/academia-app/vendor/bundle/ruby/2.2.0/gems/rake-11.3.0/exe/rake:27:in `<top (required)>'
/opt/circleci/.rvm/gems/ruby-2.2.3/gems/bundler-1.14.6/lib/bundler/cli/exec.rb:74:in `load'
/opt/circleci/.rvm/gems/ruby-2.2.3/gems/bundler-1.14.6/lib/bundler/cli/exec.rb:74:in `kernel_load'
/opt/circleci/.rvm/gems/ruby-2.2.3/gems/bundler-1.14.6/lib/bundler/cli/exec.rb:27:in `run'
/opt/circleci/.rvm/gems/ruby-2.2.3/gems/bundler-1.14.6/lib/bundler/cli.rb:335:in `exec'
/opt/circleci/.rvm/gems/ruby-2.2.3/gems/bundler-1.14.6/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
/opt/circleci/.rvm/gems/ruby-2.2.3/gems/bundler-1.14.6/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
/opt/circleci/.rvm/gems/ruby-2.2.3/gems/bundler-1.14.6/lib/bundler/vendor/thor/lib/thor.rb:359:in `dispatch'
/opt/circleci/.rvm/gems/ruby-2.2.3/gems/bundler-1.14.6/lib/bundler/cli.rb:20:in `dispatch'
/opt/circleci/.rvm/gems/ruby-2.2.3/gems/bundler-1.14.6/lib/bundler/vendor/thor/lib/thor/base.rb:440:in `start'
/opt/circleci/.rvm/gems/ruby-2.2.3/gems/bundler-1.14.6/lib/bundler/cli.rb:11:in `start'
/opt/circleci/.rvm/gems/ruby-2.2.3/gems/bundler-1.14.6/exe/bundle:32:in `block in <top (required)>'
/opt/circleci/.rvm/gems/ruby-2.2.3/gems/bundler-1.14.6/lib/bundler/friendly_errors.rb:121:in `with_friendly_errors'
/opt/circleci/.rvm/gems/ruby-2.2.3/gems/bundler-1.14.6/exe/bundle:24:in `<top (required)>'
/opt/circleci/.rvm/gems/ruby-2.2.3/bin/bundle:23:in `load'
/opt/circleci/.rvm/gems/ruby-2.2.3/bin/bundle:23:in `<main>'
/opt/circleci/.rvm/gems/ruby-2.2.3/bin/ruby_executable_hooks:15:in `eval'
/opt/circleci/.rvm/gems/ruby-2.2.3/bin/ruby_executable_hooks:15:in `<main>'
(See full trace by running task with --trace)

bundle exec rake db:create db:structure:load returned exit code 1

Action failed: bundle exec rake db:create db:structure:load
```

**Comment by camertron** [comment]
@patbl what's the value of `config.assets.compile` in the Rails environment you're running in? As of 4.2, Rails won't create a sprockets environment if `config.assets.compile` is set to `false`. If generated-assets detects that compilation has been turned off, it won't perform the setup necessary to add i18n files to the asset precompile path, and thus you'll see outdated translations.

**Comment by patbl** [comment]
`config.assets.compile` is `true` in the development environment, which is where I noticed the translations not being updated.

**Comment by camertron** [comment]
@patbl hmm this is really interesting/strange. In my test rails 4.2.8 app it looks like "hot" reloading doesn't work. In other words, as you described, changes to the translation .yml files don't show up in the browser (you'll see the old translations). I'm a bit baffled that hot reloading worked for you under Rails 4.2.7.1, because I couldn't reproduce that behavior in my tester app. The only way I can get it to work is by running `rake tmp:cache:clear`.

The error you got when downgrading to an earlier version of i18n-js-assets is likely due to the Rails 5 compatibility fix, which appears to also affect the latest release of the 4.2 branch (i.e. 4.2.8). So, you definitely want to use the most recent version if i18n-js-assets.

To that end, I did a little research and discovered that sprockets allows assets to depend on each other. If a dependent asset or file changes, the parent asset will be recompiled. It seems natural therefore to have .i18njs files depend on the .yml files in the I18n load path. Here's the PR I created to introduce the dependency. Would you mind adding it to your Gemfile to see if it fixes your reloading issues? In your Gemfile:

```ruby
gem 'i18n-js-assets', github: 'camertron/i18n-js-assets', branch: 'dependencies'
```

Thanks!

**Comment by patbl** [comment]
Hot reloading on that branch works with 4.2.7.1 and 4.2.8. Thanks!

**Comment by camertron** [comment]
Awesome! Published as v2.2.0.
